src/main/kotlin/com/example/delivery/repository/impl/OrderRepositoryImpl.kt

```diff
@@ -45,8 +118,8 @@ class OrderRepositoryImpl(var mongoTemplate: MongoTemplate) : OrderRepository {
         return mongoTemplate.findAndModify(query, update, FindAndModifyOptions.options().returnNew(true), className)
     }
 
-    override fun findAll(id: String): List<MongoOrder> {
-        val query = Query.query(Criteria.where(MongoOrder::userId.name).isEqualTo(id))
-        return mongoTemplate.find(query, className)
+    override fun fetchProducts(productIds: List<String>): List<MongoProduct> {
+        val query = Query(Criteria.where("_id").`in`(productIds))
+        return mongoTemplate.find(query, MongoProduct::class.java)
```

Comment:
You can use Spring Data MongoDB extensions with reified, in this case:
```suggestion
        return mongoTemplate.find<MongoProduct>(query)
```